src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -1407,5 +1427,251 @@ End Class
 End Namespace";
             await VerifyVB.VerifyAnalyzerAsync(code);
         }
+
+        [Fact]
+        public async Task TestEnumeratedParameterAfterLinqCallChain()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> i, IEnumerable<int> j, IEnumerable<int> k)
+    {
+        var z = k.Concat(j).Concat(i);
+        [|j|].ElementAt(10);
+        [|z|].ToArray();
+        [|k|].ToArray();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(code);
+        }
+
+        [Fact]
+        public async Task TestEnumeratedLocalAfterLinqCallChain1()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> i, IEnumerable<int> k)
+    {
+        var j = Enumerable.Range(1, 10);
+        var z = i.Concat(j).Concat(k);
+        [|j|].ElementAt(10);
+        [|z|].ToArray();
+        [|k|].ToArray();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(code);
+        }
+
+        [Fact]
+        public async Task TestEnumeratedLocalAfterLinqCallChain2()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> i, IEnumerable<int> k)
+    {
+        var j = Enumerable.Range(1, 10).Except(i).Except(k);
+        var z = i.Concat(k);
+        [|j|].ElementAt(10);
+        [|z|].ToArray();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(code);
+        }
+
+        [Fact]
+        public async Task TestEnumeratedLocalAfterLinqCallChain3()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> i)
+    {
+        var j = Enumerable.Range(1, 10);
+        var p = j;
+        var z = i.Concat(j).Concat(p);
+        [|j|].ElementAt(10);
+        [|z|].ToArray();
```

Comment:
I am not sure about the correct way to report diagnostic here:
here `z` enumerates 2 collections
1. The result of `Enumerable.Range(1, 10);` on line 1466
2. The result of `i.Concat(j).Concat(k);`

And for this test, it needs to flag 1 and 2.
But not we are just reporting 'Possible multiple enumerations of 'IEnumerable' collection in this method'
Should I change the message to "Possible multiple enumerations of 'IEnumerable' collection '{0}' in this method" and report multiple diagnostic under the same operation? 
Is it OK to report multiple diagnostic with the same title under the same location?

@mavasani